core/src/utils/native/haptic.ts

```diff
@@ -28,37 +94,34 @@ const HapticEngine = {
      * the Vibrate API. This check avoids that error
      * if the browser does not support the Vibrate API.
      */
-    if (win.Capacitor?.getPlatform() === 'web') {
+    if (capacitor?.getPlatform() === 'web') {
       return typeof navigator !== 'undefined' && navigator.vibrate !== undefined;
     }
 
     return true;
   },
   isCordova() {
-    return !!(window as any).TapticEngine;
+    return (window as any).TapticEngine !== undefined;
   },
   isCapacitor() {
-    const win = window as any;
-    return !!win.Capacitor;
+    return getCapacitor() !== undefined;
   },
   impact(options: HapticImpactOptions) {
     const engine = this.getEngine();
     if (!engine) {
       return;
     }
-    const style = this.isCapacitor() ? options.style.toUpperCase() : options.style;
-    engine.impact({ style });
+    engine.impact({ style: options.style });
```

Comment:
The Cordova plugin uses all lowercase keywords, so we might need to do `toLowerCase()` for compatibility: https://github.com/EddyVerbruggen/cordova-plugin-taptic-engine

We should also add a tech debt ticket to remove that support since the other native integrations we used only support Capacitor.

Reply:
Re-added the fallback behavior here: https://github.com/ionic-team/ionic-framework/pull/27755/commits/a2339cbfd8afdc7b1e6cb4afb451a89a9335cc9d

Also created a ticket to track removing the legacy Cordova integrations for the haptics plugin.